vulnerabilities/xxe/index.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+define( 'DVWA_WEB_PAGE_TO_ROOT', '../../' );
+require_once DVWA_WEB_PAGE_TO_ROOT . 'dvwa/includes/dvwaPage.inc.php';
+
+dvwaPageStartup( array( 'authenticated', 'phpids' ) );
+
+$page = dvwaPageNewGrab();
+$page[ 'title' ]   = 'XML external identity (XXE)' . $page[ 'title_separator' ].$page[ 'title' ];
+$page[ 'page_id' ] = 'xxe';
+$page[ 'help_button' ]   = 'xxe';
+$page[ 'source_button' ] = 'xxe';
+
+dvwaDatabaseConnect();
+
+$vulnerabilityFile = '';
+switch( $_COOKIE[ 'security' ] ) {
+	case 'low':
+		$vulnerabilityFile = 'low.php';
+		break;
+	case 'medium':
+		$vulnerabilityFile = 'medium.php';
+		break;
+	case 'high':
+		$vulnerabilityFile = 'high.php';
+		break;
+	default:
+		$vulnerabilityFile = 'impossible.php';
+		break;
+}
+
+require_once DVWA_WEB_PAGE_TO_ROOT . "vulnerabilities/xxe/source/{$vulnerabilityFile}";
+
+
+$page[ 'body' ] .= "
+<div class=\"body_padded\">
+	<h1>XML external identity (XXE)</h1>
+
```

Comment:
I'd suggest putting a sample of the type of XML content the upload is expecting in here. Otherwise, the user won't know what a valid format looks like before they start hacking it.

Maybe give them a working sample file to download.